yarn-project/noir-contracts/src/contracts/schnorr_auth_witness_account_contract/src/main.nr

```diff
@@ -19,23 +64,62 @@ contract SchnorrAuthWitnessAccount {
           payload.serialize(), 
           GENERATOR_INDEX__SIGNATURE_PAYLOAD
         )[0];
-        _inner_is_valid(message_hash, context.this_address());
+        assert(_inner_is_valid(message_hash, context.this_address()));
         payload.execute_calls(&mut context);
     }
 
     #[aztec(private)]
     fn is_valid(
         message_hash: Field
     ) -> Field {
-        _inner_is_valid(message_hash, context.this_address());
-        0xe86ab4ff
+        if (_inner_is_valid(message_hash, context.this_address())){
+            0xe86ab4ff
+        } else {
+            0
+        }
+    }
+
+    #[aztec(public)]
+    fn is_valid_public(
+        message_hash: Field,
+    ) -> Field {
+        let storage = Storage::init(Context::public(&mut context));
+        let value = storage.approved_action.at(message_hash).read();
+        if (value == 1){
+            0xe86ab4ff
+        } else {
+            0
+        }
+    }
+
+    #[aztec(private)]
+    fn set_is_valid_storage(
+        message_hash: Field,
+        value: Field,
+    ) {
+        assert((value == 0) | (value == 1), "value must be a boolean");
```

Comment:
are there test cases to assert this, sorry if i missed them when reading e2e_token_contract

Reply:
No, there are not atm. It will be replaced with a proper bool when you fix #2175 😎 Should probably add it in there.